src/PolyhedraEndcapCalorimeter2_geo.cpp

```diff
@@ -75,7 +76,7 @@ static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector s
       sliceZ += s_thick / 2;
       PlacedVolume s_phv = l_vol.placeVolume(s_vol, Position(0, 0, sliceZ));
       s_phv.addPhysVolID("slice", s_num);
-      if (x_slice.isSensitive()) {
+      if (x_slice.isSensitive() || allSensitive == 1) {
```

Comment:
Avoid bool to int conversion.
```suggestion
      if (x_slice.isSensitive() || allSensitive) {
```